include/swift/SIL/SILCloner.h

```diff
@@ -748,9 +748,13 @@ SILCloner<ImplClass>::visitAllocStackInst(AllocStackInst *Inst) {
     Loc = MandatoryInlinedLocation::getAutoGeneratedLocation();
     VarInfo = None;
   }
-  recordClonedInstruction(Inst, getBuilder().createAllocStack(
-                                    Loc, getOpType(Inst->getElementType()),
-                                    VarInfo, Inst->hasDynamicLifetime()));
+  auto *NewInst =
+      getBuilder().createAllocStack(Loc, getOpType(Inst->getElementType()),
+                                    VarInfo, Inst->hasDynamicLifetime());
+  // Remap the auxiliary debug scope, if any
+  if (VarInfo && VarInfo->Scope)
+    NewInst->setDebugVarScope(getOpScope(VarInfo->Scope));
```

Comment:
Should this be factored into a helper function that does this for all instructions that carry a SILDebugVariable?
I.e., `remapVarInfo(NewInst, VarInfo);`